### ISSUE TITLE
Bug:  React TS2322: Type '"fa-regular fa-arrow-down"' is not assignable to type 'IconProp'.

### ISSUE THREAD

**Issue by ShuhaoZQGG** [FortAwesome/Font-Awesome#19063]
### Bug description

(property) FontAwesomeIconProps.icon: IconProp
Type '"fa-regular fa-arrow-down"' is not assignable to type 'IconProp'.ts(2322)
index.d.ts(25, 3): The expected type comes from property 'icon' which is declared here on type 'IntrinsicAttributes & FontAwesomeIconProps'

My code:
`
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
`
`<FontAwesomeIcon icon="fa-regular fa-arrow-down" />
`
package.json:
    "@fortawesome/fontawesome-svg-core": "^6.1.1",
    "@fortawesome/free-regular-svg-icons": "^6.1.1",
    "@fortawesome/free-solid-svg-icons": "^6.1.1",
    "@fortawesome/react-fontawesome": "^0.1.18",

### Screenshots

![image](https://user-images.githubusercontent.com/20462800/169732896-dd857eab-ea08-4f7a-9412-c2e4513c4414.png)
![image](https://user-images.githubusercontent.com/20462800/169732983-8a17a555-8052-4e37-95e0-1c09d50ddecb.png)

### Font Awesome version

v6.1.1

### Serving

Self-hosted

### Implementation

SVG+JS

### Browser and Operating System

Chrome Version 101.0.4951.67 (Official Build) (64-bit)
 on Windows 10

### Web bug report checklist

- [ ] I have included a test case because my odds go _way_ up that the team can fix this when I do
- [X] I have [searched for existing issues](https://github.com/FortAwesome/Font-Awesome/issues) and to the best of my knowledge this is not a duplicate

**Comment by olmobrutall** [comment]
I'm also intereseted in a solution for this. 

This specially annoying because the sintax "fa-familyname fa-iconname" is used in the website when selecting react! (example 
https://fontawesome.com/icons/clock?s=solid&f=classic)

I think using https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html could help here: 

```TS
//Change @fortawesome\fontawesome-svg-core\index.d.ts

type ComplexIconName = `${CssStyleClass} fa-${IconName}`;  //Add line
type IconProp = IconName | [IconPrefix, IconName] | IconLookup | ComplexIconName;  //Replace line
```
With this change there will be no compile time errors... and even auto-complete works perfectly in VS Code!

![image](https://user-images.githubusercontent.com/720953/197869067-a0d5e032-c38e-4a88-970a-ce014f83ee6c.png)

The experience in Visual Studio 2022 is not so great however, looks like the auto-complete list is only considering the last token, so when you select it replaces everithing by "check"

![image](https://user-images.githubusercontent.com/720953/197869267-2af44615-7c47-44cb-b760-45023bf2ede4.png)

But I think this is definitely a VS 2022 error, not a type definition error.

**Comment by tagliala** [comment]
Hi @olmobrutall, thanks for the feedback

I'm not familiar with VSCode/VS. Is it possible that #19398 is a duplicate of this issue?

**Comment by olmobrutall** [comment]
I don't think so. This is an error for the vue type definitions. I'm using React type definitions here.

**Comment by sipi41** [comment]
Having the same problem... please help
